buildcheck/backend/yolo_processor.py

```diff
@@ -0,0 +1,167 @@
+from ultralytics import YOLO
+import cv2
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+from vectorization import *
+import shapely.geometry as geom
+
+class YOLOProcessor:
+    def __init__(self, image_path: str, model_path: str, layout: Layout):
+        self.image_path = image_path
+        self.layout = layout
+        self.model = YOLO(model_path)
+        
+    @staticmethod
+    def map_class_to_category(class_name: str) -> Category:
+        # Map YOLO detected class names to vectorization class Category enum        
+        class_mapping = {
+            'column': Category.COLUMN,
+            'curtain Wall': Category.WALL,
+            'door': Category.DOOR,
+            'railing': Category.RAILING,
+            'sliding door': Category.DOOR,
+            'stair case': Category.STAIRS,
+            'wall': Category.WALL,
+            'window': Category.WINDOW,
+        }
+        # Return mapped category or default to WALL if not found
+        return class_mapping.get(class_name, Category.WALL)
+    
+    def calculate_bbox_center(self, bbox: list) -> geom.Point:
+        x1, y1, x2, y2 = bbox
+        center_x = (x1 + x2) / 2
+        center_y = (y1 + y2) / 2
+        return geom.Point(center_x, center_y)
+    
+    def create_symbol_from_detection(self, class_name: str, bbox: list) -> Symbol:
+        x1, y1, x2, y2 = bbox
+        
+        # Create BBox object with four corner points
+        bbox_obj = BBox(
+            a=Point(x1, y1),  # Top-left
+            b=Point(x2, y1),  # Top-right
+            c=Point(x2, y2),  # Bottom-right
+            d=Point(x1, y2)   # Bottom-left
+        )
+        
+        # Map class name to category
+        category = self.map_class_to_category(class_name)
+        
+        # Create symbol
+        symbol = Symbol(category, bbox_obj)
+        
+        return symbol
+    
+    def find_room_for_symbol(self, center_point: geom.Point) -> Room:
+        for room in self.layout.rooms:
+            # Create a polygon for each room
+            room_polygon = geom.Polygon([(p.x, p.y) for p in room.junctions])
+            
+            if room_polygon.contains(center_point):
+                return room
```

Comment:
We may want to be more general. For instance, many times, the symbol bbox for a door is on the outer edge of the room polygon, and its center is definitely not within.

What I'm thinking is that we can create a `shapely.box` from the bbox and use `shapely.intersection` to compute the intersection polygon. Then, we can say it should be included if the area is above a certain threshold.